src/components/calcite-pick-list/shared-list-logic.ts

```diff
@@ -79,7 +82,8 @@ export const sharedListMethods = {
       item.icon = this.getIconType();
       item.compact = this.compact;
       if (item.selected) {
-        updateSelectedValuesMap(this.selectedValues, item);
+        this.selectedValues.set(item.value, item as any);
```

Comment:
nitpick: Can we give it a type here so that it's not using any?

Reply:
this is the alternative fix that allowed to remove the behemoth boilerplate function. That's the alternative, if you prefer that I can go back to that. IMO it hurts readability and is just a workaround to please the compiler.

Reply:
I agree that it sucks. @jcfranco is there an easier typing alternative?

Reply:
>  remove the behemoth boilerplate function.

IMO, the helper function isn't a behemoth. Please restore. I'd like us to be as type safe as possible even if it does mean adding some extra code. https://github.com/Esri/calcite-app-components/issues/492 may help simplify some of this.

Reply:
Think I found a better solution. I was able to use the Union Type
`type pickOrValueListItem = HTMLCalcitePickListItemElement | HTMLCalciteValueListItemElement;` once I set `textHeading` to optional in pickList. Once that prop became optional they were 'alike' enough to satisfy the compiler, yay.

Reply:
https://github.com/Esri/calcite-app-components/pull/581/files#diff-e6d4d9d31a2a4213f2b8bd1ebbecd98eR71
https://github.com/Esri/calcite-app-components/pull/581/files#diff-e6d4d9d31a2a4213f2b8bd1ebbecd98eR104